model/lifecycle.go

```diff
@@ -1237,6 +1237,7 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 		Requester:           v.Requester,
 		Version:             v.Id,
 		Revision:            v.Revision,
+		MustHaveResults:     util.IsPtrSetToTrue(project.GetSpecForTask(buildVarTask.Name).MustHaveResults),
```

Comment:
why don't you just pass MustHaveResults through to the BuildVarTask during construction like we do for the other options?

Reply:
@john-m-liu I'm moving this back into the thread for consistency.

> That was because this behavior doesn't make sense to have as specific to a type of file format, since presumably any task could want to fail if it has no test results

I don't really understand what you mean.  buildVarTask here is a BuildVariantTaskUnit, which is the same for any task. To plumb this through to the buildVariantTaskUnit you would just add it in[ here](https://github.com/evergreen-ci/evergreen/blob/477b9da3b8de3259bfbf1cc038b478d77e08c81f/model/project_parser.go#L882).

Reply:
Sorry that comment was in response to Brian. I didn't add the field to the BuildVariantTaskUnit because it's not something I would think of as specific to a task+variant combo. Piping it through would make this line cleaner, but would make things messier overall since we'll be maintaining the same data in 2 spots

Reply:
If it's something that's defined for the task of a variant then it is specific to that task+variant combo (as much as all of the fields in that struct are), so I think it does belong piped through.

Reply:
That's the reverse of how I think of it. It should be defined for a task in a variant if it's something that we need to track on a variant level (ie. if there are variant overrides). We definitely have not been consistent with that over the years which is why there are so many fields in the BuildVariantTaskUnit that shouldn't necessarily be variant-specific. Also it'll allow for the option of setting it at the variant-level which I think is more specific that a user would think of it, especially since this field likely ties with the commands field which is task-specific